doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,534 @@
+  parsing output easier.
+
+#### `new-tab`
+
+`new-tab [--initialPosition x,y]|[--maximized]|[--fullscreen] [--initialRows rows] [--initialCols cols] [terminal_parameters]`
+
+Opens a new tab with the given customizations. On it's _first_ invocation, also
+opens a new window. Subsequent `new-tab` commands will all open new tabs in the
+same window.
+
+**Parameters**:
+* `--initialPosition x,y`: Create the new Windows Terminal window at the given
+  location on the screen in pixels. This parameter is only used when initially
+  creating the window, and ignored for subsequent `new-tab` commands. When
+  combined with any of `--maximized` or `--fullscreen`, TODO: what do?
+* `--initialRows rows`: Create the terminal window with `rows` rows (in
+  characters). If omitted, uses the value from the user's settings. This
+  parameter is only used when initially creating the window, and ignored for
+  subsequent `new-tab` commands. When combined with any of `--maximized` or
+  `--fullscreen`, TODO: what do?
+* `--initialCols cols`: Create the terminal window with `cols` cols (in
+  characters). If omitted, uses the value from the user's settings. This
+  parameter is only used when initially creating the window, and ignored for
+  subsequent `new-tab` commands. When combined with any of `--maximized` or
+  `--fullscreen`, TODO: what do?
+* `[terminal_parameters]`: See [[terminal_parameters]](#terminal_parameters).
+
+
+#### `split-pane`
+
+`split-pane [--target,-t target-pane] [-h]|[-v] [--percent,-P split-percentage] [terminal_parameters]`
+
+<!-- `--percent,-P` is pretty close to `-p` from `--profile`. Is this okay? -->
+
+Creates a new pane by splitting the given pane vertically or horizontally.
+
+**Parameters**:
+* `--target,-t target-pane`: Creates a new split in the given `target-pane`.
+  Each pane has a unique index (per-tab) which can be used to identify them. If
+  omitted, defaults to `0` (the first pane).
+* `-h`, `-v`: Used to indicate which direction to split the pane. `-v` is
+  "vertically" (think `[|]`), and `-h` is "horizontally" (think `[-]`). If
+  omitted, defaults to vertical. If both `-h` and `-v` are provided, defaults to
+  vertical.
+* `--percent,-P split-percentage`: Designtates the amount of space that the new
+  pane should take, as a percentage of the parent's space. If omitted, the pane
+  will take 50% by default.
+* `[terminal_parameters]`: See [[terminal_parameters]](#terminal_parameters).
+
+
+
+#### `[terminal_parameters]`
+
+Some of the preceeding commands are used to create a new terminal instance.
+These commands are listed above as accepting `[terminal_parameters]` as a
+parameter. For these commands, `[terminal_parameters]` can be any of the
+following:
+
+`[--profile,-p profile-name]|[--guid,-g profile-guid] [--startingDirectory,-d starting-directory] [commandline]`
+
+* `--profile,-p profile-name`: Use the given profile to open the new tab/pane,

```

Comment:
Just wondering, how hard would it be/ is it possible to give users the ability to tab through all of their profiles?

Reply:
This might be impossible for cmd, though I'm not totally sure how powershell's autocomplete works. I know there's magic in linux where tools can install their own autocompletion scripts, so `git checkout <TAB>` can work it's magic, but I think this is _effectively_ impossible on Windows

Reply:
It's effectively impossible for CMD, but powershell lets you do very similar completer shenanigans.

Reply:
> I know there's magic in linux where tools can install their own autocompletion scripts

That ”magic“ is&nbsp;just an&nbsp;auto‑loaded shell&nbsp;script.

**PowerShell** does&nbsp;the&nbsp;exact same&nbsp;thing, except&nbsp;with a&nbsp;callback script&nbsp;block passed to&nbsp;[`Regsiter‑ArgumentCompleter`](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/register-argumentcompleter).

Reply:
@DHowett-MSFT is that something we could ship with our app package, or would it have to be something that people load into their powershell profile manually?

Reply:
I'm going to add a note on `Regsiter‑ArgumentCompleter` to Future Considerations. Thanks!